src/lepton/jpgcoder.cc

```diff
@@ -3024,21 +3025,17 @@ bool decode_jpeg(const std::vector<std::pair<uint32_t, uint32_t> > & huff_input_
 
             // (re)set rst wait counter
             rstw = rsti;
-            if (cs_cmpc != colldata.get_num_components()) {
-                if (!g_allow_progressive) {
-                    custom_exit(ExitCode::PROGRESSIVE_UNSUPPORTED);
-                } else {
-                    is_baseline = false;
-                }
-            }
 
-            if (jpegtype != 1) {
-                if (!g_allow_progressive) {
-                    custom_exit(ExitCode::PROGRESSIVE_UNSUPPORTED);
+            if (cs_cmpc != colldata.get_num_components() || jpegtype != 1) {
+                if (g_allow_progressive) {
```

Comment:
What is difference between `g_allow_progressive` and `g_progressive_image`?

Reply:
g_allow_progressive = global setting (option !-rejectprogressive), allowance to process progressive-encoded images
g_progressive_image = is current image progressive?

Reply:
Moreover, there is third variable jpegtype:
0 - image type isn't yet known
1 - progressive
2 - baseline

So, my g_progressive_image duplicates it. I may even spend some time fixing this duplication of vars

Reply:
Please fix duplication of vars or add TODO and create task.

Reply:
The interaction between variables turned out to be more complicated. I streamlined it in later commit (not in this PR).

Reply:
Ok, at least add TODO.

Reply:
I think it's not required. Each var has its own meaning, and this code works as expected. Later commit fixed subtle bug and I will add there comment describing the interaction between vars.